src/components/View/index.js

```diff
@@ -1,45 +1,97 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './style.css';
 import PropTypes from 'prop-types';
-import { setTepmerature, setOpenWeatherIcon } from '../../utils';
+import TypeSelect from '../TypeSelect';
+import { setTemperature, setOpenWeatherIcon } from '../../utils';
+
+class View extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+    };
+
+    this.handleDrop = this.handleDrop.bind(this);
+    this.handleDragOver = this.handleDragOver.bind(this);
+    this.getTemperatureType = this.getTemperatureType.bind(this);
+  };
+
+  getTemperatureType({ target: { value } }) {
+    this.setState({ value });
+  };
+
+  handleDragOver(event) {
+    event.preventDefault();
+  }
+
+  handleDrop(event) {
+    event.dataTransfer.setData("region", this.props.data.region);
+  }
+
+  render() {
+    const { value } = this.state;
+    const {
+      data: {
+        temp,
+        icon,
+        region,
+        temp_min,
+        temp_max,
+        humidity,
+        feels_like,
+        description,
+      },
+    } = this.props;
 
-function View ({ value, teperature, description, feels_like, temp_min, temp_max, humidity, icon }){
     return (
-      <>
+      <div
+        draggable
+        className='view_board'
+        data-region={region}
+        onDrop={this.handleDrop}
+        onDragOver={this.handleDragOver}
+      >
+        <button onClick={() => this.props.removeView(region)}> Close </button>
+        <TypeSelect getTemperatureType={this.getTemperatureType} value={value} />
         <div className='view'>
           <div>
+            {region}
+            <br />
             Weather
-            <br/>
-            {setTepmerature(teperature,value)}
+            <br />
+            {setTemperature(temp, value)}
```

Comment:
here setTemperature means you give value and expect that to be set somewhere.
Change it to getFormatedTemperature or formatTemperature